mcs/class/System/System.Net.Sockets/Socket.cs

```diff
@@ -972,7 +972,9 @@ public IAsyncResult BeginConnect (string host, int port, AsyncCallback requestCa
 			if (is_listening)
 				throw new InvalidOperationException ();
 
-			return BeginConnect (Dns.GetHostAddresses (host), port, requestCallback, state);
+			IPAddress[] hostAddresses = Dns.GetHostAddressesAsync (host).GetAwaiter ().GetResult ();
```

Comment:
Unfortunately this doesn't really solve the problem. The `GetResult ()` call makes is synchronous again.

Reply:
Thanks for help!